src/nvim/msgpack_rpc/server.c

```diff
@@ -110,8 +110,8 @@ char *server_address_new(void)
 ///          on failure to bind or connect.
 int server_start(const char *endpoint)
 {
-  if (endpoint == NULL) {
-    ELOG("Attempting to start server on NULL endpoint");
+  if (endpoint == NULL || endpoint[0] == '\0') {
+    ELOG("The endpoint musn't be NULL or empty.");
     return 1;
```

Comment:
> 0 on success, 1 on a regular error, and negative errno on failure to bind or listen.

I'd argue that it should be treated as regular error, since it's input validation that happens before anything else. It's no "network error".